x/posts/internal/types/post.go

```diff
@@ -224,7 +230,7 @@ func (p Post) Equals(other Post) bool {
 		p.Subspace == other.Subspace &&
 		equalsOptionalData &&
 		p.Creator.Equals(other.Creator) &&
-		p.Medias.Equals(other.Medias)
+		p.Medias.Equals(other.Medias) && p.PollData.Equals(other.PollData)
```

Comment:
This line might throw an error if `p.PollData` is `nil`.

Reply:
I'm gonna add more tests